### PR TITLE
Misc fixes and edit Tenant dialog

### PR DESCRIPTION
**Pull request by xantorres**
* Styling adjustments
* Fix warning about "compact" prop
* Fix misc ListTable issues
* Edit Tenant feature

### CODE REVIEW

File: src/app/plugins/kubernetes/components/userManagement/tenants/actions.js [hunk 1]
```diff
@@ -13,33 +15,84 @@ export const filterValidTenants = tenant => !reservedTenantNames.includes(tenant
...
+    const updateTenantPromise = keystone.updateProject(tenantId, {
+      ...currentTenant,
+      name,
+      description,
+    })
+    const prevRoleAssignments = prevRoleAssignmentsArr.reduce((acc, roleAssignment) => ({
+      ...acc,
+      [pathStr('user.id', roleAssignment)]: pathStr('role.id', roleAssignment),
+    }), {})
+    const updateUserRolesPromises = keys({ ...prevRoleAssignments, ...roleAssignments }).reduce(
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/userManagement/tenants/actions.js]: This logic is a bit hard to follow what is going on here.  Can this be broken down more into smaller functions or approached a different way?

Also, if you are just appending to `operations` each time, would a `map` make more sense here?

Comment (xantorres) [reply src/app/plugins/kubernetes/components/userManagement/tenants/actions.js]: Not all times, in case there is nothing to update I won't add any promise to the accumulator.
This logic is convoluted because is really a bit complex, there's no way around this, I have to add/remove users and roles individual depending on what's selected. I will try to improve variable naming and split it into small functions though.
